### ISSUE TITLE
Limitations of the .interactive API

### ISSUE THREAD

**Issue by jbednar** [holoviz/hvplot#531]
The new .interactive API is really fun to work with, but I'm bumping up against either its limits or my own imagination. E.g. if I start with this example that works well:

```python
import panel as pn, xarray as xr, holoviews as hv, hvplot.xarray
pn.extension()

ds = xr.tutorial.open_dataset('air_temperature')
image = ds.hvplot('lon', 'lat')
stream = hv.streams.Tap(source=image, x=-88+360, y=40)
timeseries = ds.interactive.sel(lon=stream.param.x, lat=stream.param.y,
                                method="nearest").hvplot('time')

pn.Column(image, timeseries.dmap())
```
![image](https://user-images.githubusercontent.com/1695496/101222225-f6268280-364e-11eb-8e72-feb9926c5501.png)

That's great, but then I want to put some indication of where I tapped, as a red dot in the first plot and as numeric values in the second plot. I can't quite see how to do it. A start, with nothing actually connected:

```python
import panel as pn, xarray as xr, holoviews as hv, hvplot.xarray
pn.extension()

ds = xr.tutorial.open_dataset('air_temperature')
image = ds.hvplot('lon', 'lat')
stream = hv.streams.Tap(source=image, x=-88+360, y=40)
point = hv.Points([(260,40)]).opts(size=5, color="red")
timeseries = ds.interactive.sel(lon=stream.param.x, lat=stream.param.y,
                                method="nearest").hvplot('time').relabel(f"lon: {stream.x}, lat: {stream.y}")

pn.Column(image*point, timeseries.dmap())
```
![image](https://user-images.githubusercontent.com/1695496/101222653-d479cb00-364f-11eb-8646-75d10ef95d72.png)

I could write a DynamicMap returning a Points object, but that's an entirely new concept, so I'm wondering whether there is some .apply or other more reactive approach that fits better with the rest of the code here, i.e. without an explicit callback. 

Also not sure how to specify the title properly in the second plot; stream.param.x won't work, as it just shows the Parameter object itself rather than fetching the current value. Is there a clean way?

It also might be nice to have an automatic way of getting the default title for the second plot, via hvPlot and .interactive somehow connecting up widget values to the title the same way NdOverlays do, but I guess that would be a separate issue.

**Comment by jbednar** [comment]
Also, will `dim()` expressions work with .interactive()? If so, they should presumably be in the https://hvplot.holoviz.org/user_guide/Interactive.html docs. If not, can they?

**Comment by maximlt** [comment]
I really like this example! 

The way to customize the plot title is to give to `title` a bound function. Credits goes to Philipp for that, I remember him showing that at PyData Berlin this year:
```python
import panel as pn, xarray as xr, holoviews as hv, hvplot.xarray
pn.extension()

def title(x, y):
    return f'lon={x:.2f}, lat={y:.2f}'

ds = xr.tutorial.open_dataset('air_temperature')
image = ds.hvplot('lon', 'lat')
stream = hv.streams.Tap(source=image, x=-88+360, y=40)
point = hv.Points([(260,40)]).opts(size=5, color="red")
timeseries = ds.interactive.sel(lon=stream.param.x, lat=stream.param.y,
                                method="nearest").hvplot('time', title=pn.bind(title, stream.param.x, stream.param.y))

pn.Column(image*point, timeseries.dmap())
```

My limited feedback on this example would be that:
- as a hvPlot user I'd rather not have to use HoloViews directly , or at least have one place where the streams I can use are documented. I think this is very similar to the documentation of the Panel widgets that a hvPlot user might want to inject in a pipeline. I think we said that hvPlot should document the most common widgets and then link to the Reference Gallery of Panel. There's a [Stream](https://holoviews.org/reference/index.html#streams) section in HoloViews' Reference Gallery but as streams are very useful on their own there's lots of HoloViews code in there which might be confusing to hvPlot users.
- in this particular example it'd be indeed nice to have a simple way to have a marker to show the last selected location, maybe that should be the default?

@philippjfr do you have ideas on how to improve that?

**Comment by jbednar** [comment]
I agree with all that feedback. Now I sit back and hope for it all to happen, magically!
